### PR TITLE
Add PostgreSQL privilege loader

### LINKED ISSUE #9853: Load users and privileges from PostgreSQL
Hi,

This issue is part of #8966  and to load the privileges from PostgreSQL. You can refer to `MySQLPrivilegeLoader` to finish this one. BTW, please give a look at #9851 for a better implement.

### Task
- Implement `PostgreSQLPrivilegeLoader `
- Add a test class for it (#9852 will give some hints).

Have a good day. :)

### PR DESCRIPTION
**Pull request by zhujunxxxxx**
Fixes #9853.

Changes proposed in this pull request:
- add PostgreSQL privilege loader
- add PostgreSQL privilege loader testcase

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/auth/builder/loader/dialect/PostgreSQLPrivilegeLoader.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            case "INSERT":
+                return PrivilegeType.INSERT;
+            case "UPDATE":
+                return PrivilegeType.UPDATE;
+            case "DELETE":
+                return PrivilegeType.DELETE;
+            case "CREATE":
+                return PrivilegeType.CREATE;
+            case "REFERENCES":
+                return PrivilegeType.REFERENCES;
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/auth/builder/loader/dialect/PostgreSQLPrivilegeLoader.java:117]: How about `usage`, `connect` and others?  If needed, you can add more types in `PrivilegeType`.

Comment (tristaZero) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/auth/builder/loader/dialect/PostgreSQLPrivilegeLoader.java:117]: Can we transfer `Role Attributes` (login,superuser,createdb,createrole,replication,passwod,inherit) and put them info this privilege system?

**Review by tristaZero — CHANGES_REQUESTED** [review]
Hi @zhujunxxxxx ,

I am curious there are only privileges on tables in Pg database? How about super privilege?

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/auth/builder/loader/dialect/PostgreSQLPrivilegeLoader.java (hunk 1 above)
Comment (zhujunxxxxx) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/auth/builder/loader/dialect/PostgreSQLPrivilegeLoader.java:117]: OK，I will add other `PrivilegeType`.

> I am curious there are only privileges on tables in Pg database? How about super privilege?
Can you give me some tips on how to get super privilege，I can't find it myself, it's not like MySQL.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/auth/builder/loader/dialect/PostgreSQLPrivilegeLoader.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        addToPrivilegeTypesIfPresent(resultSet.getBoolean("rolcreaterole"), PrivilegeType.CREATE_ROLE, result);
+        addToPrivilegeTypesIfPresent(resultSet.getBoolean("rolcreatedb"), PrivilegeType.CREATE_DATABASE, result);
+        addToPrivilegeTypesIfPresent(resultSet.getBoolean("rolreplication"), PrivilegeType.REPL_CLIENT, result);
+        addToPrivilegeTypesIfPresent(resultSet.getBoolean("rolinherit"), PrivilegeType.INHERIT, result);
+        addToPrivilegeTypesIfPresent(resultSet.getBoolean("rolcanlogin"), PrivilegeType.CAN_LOGIN, result);
+        return result;
+    }
+
+    private String getTablePrivilegeSQL(final Collection<ShardingSphereUser> users) {
+        String userList = users.stream().map(each -> each.getGrantee().getUsername())
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/auth/builder/loader/dialect/PostgreSQLPrivilegeLoader.java:139]: Hi which is correct one for a query? `user1, user2` or `'user1', 'user2'`?

**Review by tristaZero — COMMENTED** [review]
Hey, better!

**Review by tristaZero — APPROVED** [review]
Hi, nice job.

**Pull request merged**